### ISSUE TITLE
VPN App on Mac stalls on the "Waiting for sign in" page, after intial install and first launch.  

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#3559]
I have just purchased a MacBook Air and was trying to install Mozilla VPN on it. I currently have VPN installed on 2 devices.

1. First I went to 

[https://www.mozilla.org/en-US/products/vpn/](https://www.mozilla.org/en-US/products/vpn/|smart-link) 

2. Selected “Already a subscriber”

3. Clicked the “Download” VPN for Mac

4. Installed app in Application folder on my computer

5. Launched the app from the Applications folder on my computer.

6. The app opens and displays the following page with spinning icon. I could not get any further

!mail.png|width=641,height=1000!

7. I signed in to my firefox account on my browser but that did not trigger anything.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-2243)
┆Reporter: Diana Horn
┆Link To Issue: https://mozilla-hub.atlassian.net/browse/VPN-2243

**Comment by data-sync-user** [comment]
➤ Andrea Marchesini commented:

It would be great if you can provide:

  * client version
  * logs
  * prov vs stage

Thanks!

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

Per the log…

OS -> macos

OS Version -> 12.3

APP Version -> 2.8.3

Build ID -> 2.202205061632

Device ID > E9AA979040A7-58E7-86AF-85DF38C82592

See the Logs. ( https://docs.google.com/document/d/1Kbh_88IditTLoQMaJDOce_HZaUBrfVWvlfb2Ytzzq0E/edit?usp=sharing )

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

Andrea Marchesini  added the info

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

This is production.

**Comment by data-sync-user** [comment]
➤ Andrea Marchesini commented:

it seems that the app is unable to open the auth URL in the browser. What’s your default browser? Does it fix, if you click cancel and try again?

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

This issue happened yesterday when I was on the YMCA guest wifi. I am not able to replicate it today on my home wifi. I will wait to log in and try and replicate at the YMCA again

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

When I was at the YMCA when the sign in was not triggering, I tried closing the VPN and reopening it. I did not reinsall the app. The problem persisted.

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

Andrea Marchesini Joseph Anthony Cuevas Ok, I was able to replicate the experience consistently while connected to another public wifi (library). Although I don’t think it’s the wifi. Will do another test at home.

Here are the conditions that cause the app to stall at the “Waiting for signin…” page, and not open the browser sign in page…

Condition 1: 1st install of VPN, never signed on before

Condition 2: Computer has just been started, default brower is NOT open or active.

After the computer has just been turned on and the Firefox browser has not been opened. I open the VPN and that does not initiate the sign on browser. I can quit the VPN, cancel and restart and same thing happens.

Once I open a Firefox browser (current default browser), then it works. Even if I quit the Firefox browser and/or quit the VPN, and then open VPN it will  trigger the sign in page to reopen.

Another thing to note is that it doesn’t matter if I have a Chrome window (non default browser) open. Meaning I have to open the Firefox (default) browser to make it work.

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

Ok last test. I switched my default browser to Chrome. Replicated the restart with no active browsers, launched VPN, and the sign in page launched in Chrome as expected.

So it looks like there is something wrong with the trigger from the app to open Firefox brower (when it’s the default browser) and when there are no active Firefox browser windows open.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

We tried to reproduce the issue while using a MacBook Air with OS 12.3.1, Firefox Release 100.0.1 as default browser, and a WiFi network, but we could not. We tried different browsers as well. The only difference I might think about is that on the device we used, we had Mozilla VPN installed in the past.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Diana Horn I will say that I have seen with firefox as my browser that it can take a really long time for the browser to open an auth tab....like longer than 30s which is an eternity when you clicked a button and are waiting for something to open.

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

I replicated on my home network and it worked as expected, the sign in browser poped up quickly. I can try on yet another guest wifi to see if I experience the same problem and want wait. 

If a 30 sec wait is possible then we need to add a message to explain there might be a delay to prevent user dropoff.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Diana Horn Could be because of a previous sign out action? More details tracked here:https://mozilla-hub.atlassian.net/browse/VPN-1155 ( https://mozilla-hub.atlassian.net/browse/VPN-1155|smart-link )

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

I will try to replicate over the weekend at the location outside of my home. Valentina Virlics I had not installed or signed into FX on the new computer.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Diana Horn In 2.12 we are moving to in-app auth so I believe this problem will go away. I will ping you when it’s ready for testing.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Diana Horn would you be able to test out our 2.12 candidate build and see if you can reproduce?  https://archive.mozilla.org/pub/vpn/candidates/2.12.0-candidates/build20221202092647/mac/ ( https://archive.mozilla.org/pub/vpn/candidates/2.12.0-candidates/build20221202092647/mac/|smart-link )

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

Sarah Bird what’s the best way for Diana to do a clean re-install on macOS to try again? (In Windows, you seemingly have to manually delete C:\Users\$user\AppData\Roaming\Mozilla)

**Comment by data-sync-user** [comment]
➤ Gela Malek Pour commented:

Santiago Andrigo  the following should act as a clean reinstall:

  1. Remove the application: Go to finder → Applications → Right click and Move to Trash → Empty the trash
  2. Remove all temporary files: open Finder and click on Go → Go to Folder and go to /private/var/log and delete the folder mozillavpn
  3. Similarly go to /Users/mozilla/Library/Caches and remove Mozilla and Mozilla VPN

I would also restart the machine just in case there is anything else lingering. This should remove all previous traces of the Mozilla VPN from your machine.

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

Thanks Gela Malek Pour ! Diana Horn mind trying that?

**Comment by data-sync-user** [comment]
➤ Diana Horn commented:

Ok happy to report that I have tried many different variations and have not been able to replicate the error on the public wifi where I encountered the original bug.

For reference, here’s a summary of the attempts I took:

Attempt 1: Deleted the app, emptied trash. Installed from live site. After install it went to the app home page, used my email from cache or cookies

Attempt 2:  Deleted the app, emptied trash, did step 2 from Gella, couldn’t find folder Step 3 in the Go menu. Installed from Sarah’s link below. Again, After install it went to the app home page, used my email from cache or cookies.

Attempt 3: ✅ Signed out of VPN, Changed my sign in to FFX to another account. Deleted the app, emptied trash, did step 2 from Gella, couldn’t find folder Step 3 in the Go menu. Installed from Sarah’s link below. After install it went to the app home page, used my email from cache or cookies.

!image-20230104-231410.png|width=50%!

and then briefly paused on the screen with the circling logo that it was getting hung up on, then went to this page

!image-20230104-231437.png|width=366,height=656!

So it looks like this error has been fixed.
